docs/migration/00a_Solving_problems_with_v0.5.ipynb

```diff
@@ -0,0 +1,431 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "d62d7cec",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit_nature.circuit.library import UCC, UVCC\n",
+    "\n",
+    "ucc = UCC(qubit_converter=None, num_particles=None, num_spin_orbitals=None, excitations=None)\n",
+    "uvcc = UVCC(qubit_converter=None, num_modals=None, excitations=None)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7c3cab8c",
+   "metadata": {},
+   "source": [
+    "#### New\n",
+    "This was mismatching the API of the `HartreeFock` and `VSCF` initial states. We have aligned these APIs to look like in the following:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "f45a8afc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit_nature.second_q.circuit.library import UCC, UVCC\n",
+    "\n",
+    "ucc = UCC(num_spatial_orbitals=None, num_particles=None, excitations=None, qubit_converter=None)\n",
+    "uvcc = UVCC(num_modals=None, excitations=None, qubit_converter=None)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "43bb3a7e",
+   "metadata": {},
+   "source": [
+    "# `qiskit_nature.algorithms`\n",
+    "\n",
+    "The algorithms in Qiskit Nature have been updated to use the new `qiskit.algorithms` components which are based on the `qiskit.primitives` as of Qiskit Terra 0.22.\n",
+    "\n",
+    "For most changes to take effect, you can once again simply update your import paths from `qiskit_nature.algorithms` to `qiskit_nature.second_q.algorithms`. However, there are some details which you need to pay attention to, due to the change to the primitive-based algorithms of Qiskit Terra being used under the hood. These details are explained below."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9fcccba0",
+   "metadata": {},
+   "source": [
+    "### `qiskit_nature.algorithms.initial_points`\n",
+    "\n",
+    "This module was moved to `qiskit_nature.second_q.algorithms.initial_points`. All you need to do is update your imports."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "055d089b",
+   "metadata": {},
+   "source": [
+    "### `qiskit_nature.algorithms.pes_samplers`\n",
+    "\n",
+    "This module was **removed** without a replacement. The reason for that, is that we are working towards a driver-less design of Qiskit Nature, in which case the sampling of the potential energy surface becomes the responsibility of the classical code rather than Qiskit Nature."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3f1f4687",
+   "metadata": {},
+   "source": [
+    "### `qiskit_nature.algorithms.ground_state_solvers`\n",
+    "\n",
+    "This module was moved to `qiskit_nature.second_q.algorithms.ground_state_solvers`.\n",
+    "Besides updating your imports, you need to pay attention to the following:\n",
+    "\n",
+    "- `AdaptVQE` was migrated to Qiskit Terra: `qiskit.algorithms.minimum_eigensolver.AdaptVQE`\n",
```

Comment:
Could we state here that instead of a dedicated adapt vqe ground state solver, in migrating it to Terra, it becomes an algorithm there like VQE that is used with the GroundStateSolver.

(Made me thing about the AdaptVQE factory capability that we don;t have yet - maybe the existing factory could switch between VQE and AdaptVQE - anyway outside the scope of this guide)

Reply:
Yes, I actually have been working on an example on how to use `AdaptVQE`. I might make that a `HowTo` :+1: 

I also think that a dedicated factory may not be needed. But also thinking of `VQD` I was wondering if we can revisit our factory classes and instead make them `AnsatzFactory` that are re-usable for whatever algorithm..

But as you said, this is off-topic for here. Let's discuss this another time :+1: